controllers/apply-next/form-router.js

```diff
@@ -186,6 +210,79 @@ function initFormRouter(formModel) {
+
+    /**
+     * Route: Eligibility checker
+     */
+    router
+        .route('/eligibility/:step?')
+        .all((req, res, next) => {
+            if (res.locals.currentApplicationStatus === 'ineligible') {
+                return res.redirect(`${req.baseUrl}`);
+            }
+
+            res.locals.eligibilityQuestions = formModel.eligibilityQuestions;
+
+            if (!req.params.step || req.params.step > res.locals.eligibilityQuestions.length) {
+                return res.redirect(`${req.baseUrl}/eligibility/1`);
+            }
+
+            res.locals.currentStepNumber = parseInt(req.params.step);
+
+            // See how far this user has progressed with these questions
+            // (eg. to prevent URL tampering)
+            const currentProgress = parseInt(
+                get(req.session, `${SESSION_PREFIX}.eligibilityProgress.${res.locals.currentlyEditingId}`)
+            );
+
+            // Prevent skipping ahead
+            if (currentProgress < res.locals.currentStepNumber) {
+                return res.redirect(`${req.baseUrl}/eligibility/${currentProgress}`);
+            }
+
+            res.locals.currentStep = res.locals.eligibilityQuestions[res.locals.currentStepNumber - 1];
+            next();
+        })
+        .get((req, res) => {
+            res.render(path.resolve(__dirname, './views/eligibility'), {
+                eligibility: 'pending',
+                csrfToken: req.csrfToken(),
+                questions: res.locals.eligibilityQuestions
+            });
+        })
+        .post(async (req, res) => {
+            const stepAnswer = req.body.eligibility;
+            if (!stepAnswer || stepAnswer !== 'yes') {
+                // Mark this application as ineligible and show an error
+                await applicationsService.changeApplicationState(res.locals.currentlyEditingId, 'ineligible');
+                res.render(path.resolve(__dirname, './views/eligibility'), {
+                    eligibility: 'ineligible'
+                });
+            } else {
+                const isComplete = res.locals.currentStepNumber === res.locals.eligibilityQuestions.length;
+                const nextStepNumber = res.locals.currentStepNumber + 1;
+                if (!isComplete) {
+                    // Store their progress in their session
+                    set(
+                        req.session,
+                        `${SESSION_PREFIX}.eligibilityProgress.${res.locals.currentlyEditingId}`,
+                        nextStepNumber
+                    );
+                    res.redirect(`${req.baseUrl}/eligibility/${nextStepNumber}`);
+                } else {
+                    await applicationsService.changeApplicationState(res.locals.currentlyEditingId, 'eligible');
+                    res.render(path.resolve(__dirname, './views/eligibility'), {
+                        eligibility: 'eligible'
```

Comment:
Might be worth extracting these different statuses out into a constant?

Reply:
I thought about this but the constant would just use the same keys as the strings here so it feels a bit redundant? eg.

```
// before: eligibility: 'eligible'
//  after: eligibility: ELIGIBILITY_STATES.eligible

Reply:
Can be useful if you can get the code to throw if we reference a nonexistent state, otherwise yeah it might be redundant.